com.unity.netcode.gameobjects/Runtime/Connection/NetworkConnectionManager.cs

```diff
@@ -844,6 +844,16 @@ internal void OnClientDisconnectFromServer(ulong clientId)
             {
                 var transportId = ClientIdToTransportId(clientId);
                 NetworkManager.NetworkConfig.NetworkTransport.DisconnectRemoteClient(transportId);
+
+                try
+                {
+                    OnClientDisconnectCallback?.Invoke(clientId);
```

Comment:
Just want to understand here...

Based on the description, OnClientDisconnectCallback was getting invoked with the wrong client ID previously. I can see how this would invoke it with the correct ID... but I don't see any changes to stop the previous invocation from calling with the wrong ID... so won't this result in it getting called twice, once with the correct ID and once with the wrong one?

Reply:
The other one is not called if the host disconnects the client from its side.
The recent changes with the NetworkConnectionManager actually wouldn't invoke OnClientDisconnected under this scenario at all.